src/main/java/com/mixpanel/android/viewcrawler/EditProtocol.java

```diff
@@ -155,8 +155,22 @@ public Edit readEdit(JSONObject source) throws BadInstructionsException, CantGet
                 for (int i = 0; i < length; i++) {
                     JSONObject layout_info = args.optJSONObject(i);
                     ViewVisitor.LayoutRule params;
-                    params = new ViewVisitor.LayoutRule(layout_info.getInt("view_id"),
-                            layout_info.getInt("verb"), layout_info.getInt("anchor_id"));
+
+                    final Integer view_id = reconcileIdsInPath(-1, layout_info.getString("view_id"), mResourceIds);
+                    final Integer anchor_id;
+                    if (layout_info.getString("anchor_id").equals("0")) {
+                        anchor_id = 0;
+                    } else if (layout_info.getString("anchor_id").equals("-1")) {
+                        anchor_id = RelativeLayout.TRUE;
+                    } else {
+                        anchor_id = reconcileIdsInPath(-1, layout_info.getString("anchor_id"), mResourceIds);
+                    }
+
+                    if (view_id == null || anchor_id == null) {
+                        continue;
```

Comment:
This seems like an invalid message. While it probably isn't right to throw an exception (since we could have id names specified that we can't find for some reason, like multiple apps or versions reading the same decide messages) we should probably log when we're missing information, or we're getting directives for views that don't seem to exist.